modules/ting_field_search/ting_field_search.module

```diff
@@ -92,31 +92,32 @@ function ting_field_search_get_active_profile() {
  * Set the current search request type when a profile was activated.
  */
 function ting_field_search_set_context($request_type) {
-  $static = &drupal_static(__FUNCTION__);
-  $static = $request_type;
+  static $ting_field_search_static_fast = NULL;
+  $ting_field_search_static_fast = $request_type;
 }
 
 /**
  * If a profile is active this returns the associated ting request type.
  */
 function ting_field_search_get_context() {
-  return drupal_static('ting_field_search_set_context');
+  static $ting_field_search_static_fast;
+  return $ting_field_search_static_fast;
```

Comment:
This will not work!

The old code got the value set in ting_field_search_set_context() through drupal_static. This code just returns the static variable in this function, which will always be null?

You can still use the fast static pattern with drupal_static: https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/drupal_static/7.x (look at the last example).

Reply:
Also, I'm really not sure this change is needed at all.

It's just a simple flag that gets set. I don't think there's any expensive computation going on.

The active profile is statically cached (with the fast drupal static pattern) is this has the potential to result in a database query.

Reply:
Just tested it and the change (perhaps obviously) breaks the search profiles from ting_field_search. 

They now all return the same result!

I see 3 options:
1. Leave as it was before 
2. Use the fast static pattern with drupal_static as described in the documentation
3. Remove ting_field_search_get_context() and only have ting_field_search_set_context(). Then if you pass null it will return the context and act like ting_field_search_get_context(). A lot of other modules do it this way.

Reply:
You are right the fast static pattern only works inside the local scope.

So to keep it simple I have revert this change for now :-)